app/ts/components/DebounceInput.tsx

```diff
@@ -0,0 +1,59 @@
+import { Signal } from '@preact/signals'
+import { useRef, useState } from 'preact/hooks'
+
+type DebounceInputProps = {
+	placeholder?: string
+	delay?: number
+	onDebouncedChange: (value: string) => void
+	onValidationError: (value: string) => void
+	validate: (value: string) => boolean
+	validationError?: string
```

Comment:
```suggestion
	validate: (value: string) => string | undefined
```
Consider combining these so the `validate` function can return different errors, rather than having a single error string that is always used.  The way you have it now you can't have your validate function provide good feedback on the specific failure.

Reply:
added issue